### PR TITLE
Fix email authorization LateInitializationError

### PR DESCRIPTION
**Pull request by atavism**
Resolves https://github.com/getlantern/engineering/issues/1483

### CODE REVIEW

File: lib/account/device_linking/authorize_device_via_email_pin.dart [hunk 1]
```diff
@@ -74,6 +74,8 @@ class AuthorizeDeviceViaEmailPin extends StatelessWidget {
     try {
       context.loaderOverlay.show();
       await sessionModel.validateDeviceRecoveryCode(code);
+    //once code is validated, approve the device
+      await sessionModel.approveDevice(code);
```
Comment (jigar-f) [inline lib/account/device_linking/authorize_device_via_email_pin.dart:78]: @atavism I think we need to approve the device once the code is validated correct?

File: lib/ffi.dart [hunk 2]
```diff
@@ -109,6 +109,24 @@ Future<void> ffiRemoveDevice(String deviceId) async {
   return;
 }
 
+Future<String> ffiAuthorizeEmail(String emailAddress) async {
+  final json = await _bindings
+      .approveDevice(emailAddress.toPointerChar())
+      .cast<Utf8>()
+      .toDartString();
+  final result = BaseResponse.create()..mergeFromProto3Json(jsonDecode(json));
```
Comment (jigar-f) [inline lib/ffi.dart:117]: I think we need to check for API error here

**Comment by jigar-f** [comment]
@atavism I added few changes here.

File: lib/ffi.dart (hunk 2 above)
Comment (atavism) [reply lib/ffi.dart:117]: Oops, I forgot to pass that change last night. Thanks!!

**Comment by atavism** [comment]
> @atavism I added few changes here.
Thanks, @jigar-f! LGTM

**Pull request merged**